### DISCUSSION TITLE
throat saturation （Ordinary Percolation ）

### DISCUSSION THREAD

**Discussion by mourendong** [PMEAL/OpenPNM#1959] [Q&A] [answered]
Dear all，
 I do some research about evaporation through Ordinary Percolation. After a few steps of invasion, the throat saturation was changed. In general, the throat saturation should between 0 and 1, but the results of that were 0 or 1. Some throats may be partially filled with water, i.e., the saturation may be 0.6, just like this. Can you give me some advice on how to do this? ann some methods on how to simulate evaporation.
Thanks for your kind help.
**'pore. occupancy'**: A value between 0 and 1 indicating the fractional volume of each pore that is invaded.  If no late pore filling model was applied, then this will only be integer values (either filled or not).
**'throat.occupancy'** : The same as 'pore.occupancy' but for throats.

**Comment by jgostick** [comment, marked as answer]
From the perspective of percolation, a pore/throat can either be invaded or not, so is a Boolean value (0 or 1), for which we use the word occupancy.  The appearance of 'fractional' filling comes from applying additional physics on top of percolation.  For instance, it's possible to apply a 'late pore filling' model which make the 'saturation' of the pore < 1.  Upon initial invasion the pore may be filled to 0.5, while as the pressure is increased the corners and crevices begin to fill, so s -> 1.  These sort of models basically allow the inclusion of unresolved features and physics. 

Regarding evaporation, we do not have a lot of support/features for this application.  It's a very complicated process and we don't focus on this within our own research group.  However, you can implement this using OpenPNM by using the pore-scale models.  This means that you define a function that takes various other properties as input, then computes the saturation, and returns it.  This saturation value can then be used by an algorithm that model diffusion of water vapor or whatever you're focusing on.  The algorithm(s) will probably be the most complicated part, since heat and mass and invasion percolation are coupled.  We did [one paper on this several years](https://doi.org/10.1007/s10800-017-1126-6) ago but it was messy and time consuming, so the code never made it into openpnm.

I'm sorry that this is not a straight forward answer, but it's the best I can offer.

**Reply by mourendong** [reply]
Thanks for your advice. I'll try my best.
